test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -193,28 +359,54 @@ public void WhenMoreThanOneSolutionExistsInTheDirectoryItPrintsErrorAndUsage()
             cmd.StdOut.Should().Be($"Project `{projectPath}` added to the solution.");
             cmd.StdErr.Should().BeEmpty();
 
-            var slnFile = SlnFile.Read(Path.Combine(projectDirectory, "App.sln"));
-            var matchingProjects = slnFile.Projects
-                .Where((p) => p.Name == "Lib")
-                .ToList();
+            var slnPath = Path.Combine(projectDirectory, "App.sln");
 
-            matchingProjects.Count.Should().Be(1);
-            var slnProject = matchingProjects[0];
-            slnProject.FilePath.Should().Be(projectPath);
-            slnProject.TypeGuid.Should().Be(ProjectTypeGuids.CPSProjectTypeGuid);
-            if (!string.IsNullOrEmpty(projectGuid))
+            if (string.IsNullOrEmpty(expectedProjectGuid))
             {
-                slnProject.Id.Should().Be(projectGuid);
+                var slnFile = SlnFile.Read(slnPath);
+                var matchingProjects = slnFile.Projects
+                    .Where((p) => p.Name == "Lib")
+                    .ToList();
+
+                matchingProjects.Count.Should().Be(1);
+                var slnProject = matchingProjects[0];
+                expectedProjectGuid = slnProject.Id;
             }
 
-            var restoreCmd = new DotnetCommand()
+            var expectedSlnContents = expectedSlnContentsTemplate.Replace("__PROJECTGUID__", expectedProjectGuid);
+            File.ReadAllText(slnPath)
+                .Should().Be(expectedSlnContents);
```

Comment:
whenever comparing multi-line strings from two different sources use: BeVisuallyEquivalentTo - that's a tiny extension I wrote - lack of that is causing problems on Windows when git is replacing line endings